### PR TITLE
refs(api): Default to using SnubaSearchBackend for search. Remove unused code from  DjangoSearchBackend (SEN-181)

### PR DESCRIPTION
**Pull request by wedamija**
We're entirely using Snuba for search in production. It's time to start relying
on this in dev as well, so that we're getting the same results. Matt is working on
https://github.com/getsentry/sentry/pull/12173, which will make it easy to run Snuba locally. So
once that has been merged, we can merge this as well.

Removes unused `_query` code from `DjangoSearchBackend`, and cascades out to remove anything that is now unused due to that removal. I plan to just merge the django and snuba backends together, but this pr is already pretty big so holding off for now.

Note that tests/sentry/search/django/tests.py is copy/pasted in tests/snuba/search/test_backend.py and slightly modified. So we're not losing any important tests by just deleting it.

### CODE REVIEW

File: src/sentry/tagstore/v2/backend.py [hunk 1]
```diff
@@ -952,55 +952,6 @@ def get_group_tag_values_for_users(self, event_users, limit=100):
             )
         )
 
-    def get_group_ids_for_search_filter(
```
Comment (markstory) [inline src/sentry/tagstore/v2/backend.py]: This will conflict with #12212 but the conflict should be easily to solve on either side.

**Review by markstory — APPROVED** [review]
:clap: Yay to not having to maintain code we don't actually run.

**Comment by wedamija** [comment]
To fix this test failure I'm temporarily moving test_project_group_index.py into snuba tests. We'll need to consider how we handle Snuba tests going forward, since they're core now.

**Comment by wedamija** [comment]
> Doesn't this also require then the change the EVENTSTREAM? Otherwise, events will never be written into Snuba.

Yeah, I wasn't sure about that actually, but I think you're right... That'll work fine with your devservices tool right?

**Comment by mattrobenolt** [comment]
Yeah, it strictly is uses `SENTRY_EVENTSTREAM` now to control what needs to be run. But regardless, this is required to get data being written into Snuba. So even if the search backend is Snuba, and you are running Snuba, nothing is getting into Snuba without change there.

**Comment by wedamija** [comment]
> Yeah, it strictly is uses `SENTRY_EVENTSTREAM` now to control what needs to be run. But regardless, this is required to get data being written into Snuba. So even if the search backend is Snuba, and you are running Snuba, nothing is getting into Snuba without change there.

I added `SnubaEventStream` as the default, not clear on whether we should be using the Kafka one yet?

**Review by mattrobenolt — CHANGES_REQUESTED** [review]
For now, I’m blocking this since it has ramifications I don’t think we’re ready for just yet. Once we merge this, open source users will no longer be able to run Sentry, and all the people well indestructible to run from git, will immediately have everything broken. This will then lead to questions about how to make it work, when we don’t have any documentation or plan yet for others running Snuba. The Snuba backend chosen here is also not suitable for production and is only useful for local dev. Any production needs the Kafka based one. 

There’s also currently an outstanding issue related to Snuba and TSDB in which literally sentry core codebase does not have the needed pieces, and are still in getsentry. This means it’s impossible to get a 100% working version of Sentry with Snuba fully at the moment.

**Comment by wedamija** [comment]
> For now, I’m blocking this since it has ramifications I don’t think we’re ready for just yet. Once we merge this, open source users will no longer be able to run Sentry, and all the people well indestructible to run from git, will immediately have everything broken. This will then lead to questions about how to make it work, when we don’t have any documentation or plan yet for others running Snuba. The Snuba backend chosen here is also not suitable for production and is only useful for local dev. Any production needs the Kafka based one.
> 
> There’s also currently an outstanding issue related to Snuba and TSDB in which literally sentry core codebase does not have the needed pieces, and are still in getsentry. This means it’s impossible to get a 100% working version of Sentry with Snuba fully at the moment.

@mattrobenolt Now that we've cut the 9.1 release I'd like to push forward on this. I'm not sure what else we need to be doing here - we can't use `SnubaEventStream` in production, but it makes sense to use it in dev now. What changes are left here before we can land this and remove these old backends?

**Comment by mattrobenolt** [comment]
I am ok with this as long as we make sure the transition path for internal people is fine and local devs won't be wrecked.

**Comment by wedamija** [comment]
> I am ok with this as long as we make sure the transition path for internal people is fine and local devs won't be wrecked.

People should all be using devservices now, so I don't anticipate there being any problems. I'll get a couple people to try pulling down this branch and just running their local environment, run a couple tests etc to be sure.

**Comment by wedamija** [comment]
@mattrobenolt I've tested this locally and with @EvanPurkhiser and things look good. Mind approving/unblocking?

I'll send out a short email afterwards just letting people know. I don't think there's much that needs to be done, but I'll make sure that people unset `SENTRY_SEARCH` and `SENTRY_EVENTSTREAM` in their sentry.conf.py, and are running devservices.

**Review by mattrobenolt — APPROVED** [review]

**Pull request merged**